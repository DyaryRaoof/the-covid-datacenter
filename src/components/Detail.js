import { Row, Col } from 'react-bootstrap';
import { BsArrowRightCircle } from 'react-icons/bs';
import europe from '../images/europe-map.png';

const countries = [
  { name: 'CZECH REPUBLIC', number: 119, id: 1 },
  { name: 'ITALY', number: 119, id: 2 },
  { name: 'CZECH REPUBLIC', number: 119, id: 3 },
  { name: 'CZECH REPUBLIC', number: 119, id: 4 },
  { name: 'CZECH REPUBLIC', number: 119, id: 5 },
  { name: 'CZECH REPUBLIC', number: 119, id: 6 },
  { name: 'CZECH REPUBLIC', number: 119, id: 7 },
  { name: 'CZECH REPUBLIC', number: 119, id: 8 },
];

const Detail = () => (
  <main>
    <section className="top-section">
      <Row>
        <Col>
          <img className="w-100 opacity-25 ms-4" src={europe} alt="Europe map" />
        </Col>
        <Col>
          <div className=" top-section-content d-flex flex-column justify-content-center align-items-end me-3">
            <h1>Europe</h1>
            <span>6,958</span>
          </div>
        </Col>
      </Row>
    </section>
    <span className="stats">STATS BY COUNTRY</span>
    <section>
      <div className="town">
        {countries.map((country, index) => (
          <div className={index % 2 === 0 ? 'light-background d-flex justify-content-between align-items-center ps-3' : 'd-flex justify-content-between align-items-center ps-3'} key={country.id}>
            <h5>Prague</h5>

            <div className="d-flex  align-items-center justify-content-center text-end my-3 me-2">
              <span className="me-3">965 views</span>

              <div className="d-flex justify-content-end me-2 mt-2">
                <BsArrowRightCircle />
              </div>
            </div>
          </div>
        ))}
      </div>
    </section>
  </main>
);

export default Detail;
