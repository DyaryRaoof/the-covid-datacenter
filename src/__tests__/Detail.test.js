import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Detail from '../components/Detail';
import store from '../redux/configureStore';

describe('test Detail component', () => {
  test('should contain a section element', () => {
    render(
      <Provider store={store}>
        <Router>
          <Detail />
        </Router>
      </Provider>,
    );
    return expect(screen.getByTestId('section')).toBeInTheDocument;
  });
});

describe('test with jest snapshot', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Detail />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
