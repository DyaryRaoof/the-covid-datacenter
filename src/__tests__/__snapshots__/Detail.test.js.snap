// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with jest snapshot should render correctly 1`] = `
<main>
  <section
    className="top-section"
    data-testid="section"
  >
    <div
      className="d-flex justify-content-between"
    >
      <div>
        <img
          alt="Europe map"
          className="map opacity-25 ms-4"
          src="united-states-map.png"
        />
      </div>
      <div>
        <div
          className=" top-section-content d-flex flex-column justify-content-center align-items-end me-3"
        >
          <h1
            className="text-end"
          >
            
          </h1>
          <span>
            0
          </span>
        </div>
      </div>
    </div>
  </section>
  <span
    className="stats"
  >
    TODAYS CONFIRMED CASES
  </span>
  <section>
    <div
      className="town"
    />
  </section>
</main>
`;
